### PR TITLE
Pokemon storage system

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/pokemon_storage_system_5.c [hunk 1]
```diff
@@ -0,0 +1,2036 @@
...
+static void AddBoxMenu(void);
+static bool8 sub_8094924(void);
+static bool8 sub_809494C(void);
+static bool8 sub_8094A0C(void);
+static void sub_8094AD8(void);
+static void sub_8094C84(void);
+
+static const u16 gHandCursorPalette[] = INCBIN_U16("graphics/interface/pss_unk_83D2BCC.gbapal");
+static const u16 gHandCursorTiles[] = INCBIN_U16("graphics/interface/pss_unk_83D2BEC.4bpp");
+static const u16 gHandCursorShadowTiles[] = INCBIN_U16("graphics/interface/pss_unk_83D33EC.4bpp");
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_5.c]: should start with `s`

File: src/pokemon_storage_system_3.c [hunk 2]
```diff
@@ -0,0 +1,2779 @@
...
+#include "task.h"
+#include "text_window.h"
+#include "trig.h"
+#include "constants/items.h"
+#include "constants/help_system.h"
+#include "constants/songs.h"
+#include "constants/flags.h"
+#include "constants/vars.h"
+
+EWRAM_DATA struct PokemonStorageSystemData *sPSSData = NULL;
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_3.c]: `gPSSData`

File: src/pokemon_storage_system_3.c [hunk 3]
```diff
@@ -0,0 +1,2779 @@
...
+static void ClearBottomWindow(void);
+static void AddWallpaperSetsMenu(void);
+static void AddWallpapersMenu(u8 wallpaperSet);
+static void sub_808FDFC(void);
+static void sub_808FE54(u8 species);
+static void sub_808FF70(void);
+
+static const u32 gPokemonStorageScrollingBGTileset[] = INCBIN_U32("graphics/interface/pss_unk_83CE438.4bpp.lz");
+static const u32 gPokemonStorageScrollingBGTilemap[] = INCBIN_U32("graphics/interface/pss_unk_83CE4D0.bin.lz");
+static const u16 gPokemonStorageScrollingBGPalette[] = INCBIN_U16("graphics/interface/pss_unk_83CE5DC.gbapal");
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_3.c]: should start with `s`

File: src/pokemon_storage_system_3.c [hunk 4]
```diff
@@ -0,0 +1,2779 @@
...
+    }
+};
+
+static const struct SpritePalette gWaveformSpritePalette = {
+    gUnknown_83CE7F0, TAG_PAL_WAVEFORM
+};
+
+static const struct SpriteSheet gWaveformSpriteSheet = {
+    gUnknown_83CE810, 0x01c0, TAG_TILE_WAVEFORM
+};
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_3.c]: should start with `s`

File: src/pokemon_storage_system_3.c [hunk 5]
```diff
@@ -0,0 +1,2779 @@
...
+            sPSSData->state++;
+        }
+        else
+        {
+            SetPSSCallback(Cb_PrintCantStoreMail);
+        }
+        break;
+    case 1:
+        sub_8094D14(2);
+        Item_FromMonToMoving((sInPartyMenu != FALSE) ? CURSOR_AREA_IN_PARTY : CURSOR_AREA_IN_BOX, GetBoxCursorPosition());
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_3.c]: is ` != FALSE` here necessary for matching?

File: src/pokemon_storage_system_3.c [hunk 6]
```diff
@@ -0,0 +1,2779 @@
...
+{
+    switch (sPSSData->state)
+    {
+    case 0:
+        ClearBottomWindow();
+        sPSSData->state++;
+        break;
+    case 1:
+        sub_8094D14(2);
+        Item_GiveMovingToMon((sInPartyMenu != FALSE) ? CURSOR_AREA_IN_PARTY : CURSOR_AREA_IN_BOX, GetBoxCursorPosition());
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_3.c]: same issue here and below

File: src/pokemon_storage_system_3.c [hunk 7]
```diff
@@ -0,0 +1,2779 @@
...
+    }
+
+    DestroyTask(taskId);
+}
+
+static void GiveChosenBagItem(void)
+{
+    u16 item = gSpecialVar_ItemId;
+
+    if (item != 0)
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_3.c]: `ITEM_NONE`

File: include/pokemon_storage_system_internal.h [hunk 8]
```diff
@@ -0,0 +1,516 @@
...
+    /* 22bc */ u8 field_22C4[0x800];
+    /* 2abc */ u8 field_2AC4[0x1800];
+    /* 42bc */ u8 field_42C4[0x800];
+    /* 4abc */ u8 field_4AC4[0x1000];
+    /* 5abc */ u8 field_5AC4[0x800];
+}; // size=62bc
+
+extern struct PokemonStorageSystemData *gPSSData;
+
+void Cb2_EnterPSS(u8 a0);
```
Comment (jiangzhengwenjz) [inline include/pokemon_storage_system_internal.h]: Do you think we can standardize `Cb` and `CB`

File: src/pokemon_storage_system_2.c [hunk 9]
```diff
@@ -0,0 +1,653 @@
...
+    struct SpriteTemplate template;
+    struct OamData oamData = {};
+    oamData.size = SPRITE_SIZE(64x64);
+    oamData.paletteNum = 1;
+    template = (struct SpriteTemplate){
+        0, 0, &oamData, gDummySpriteAnimTable, NULL, gDummySpriteAffineAnimTable, SpriteCallbackDummy
+    };
+    {
+    const u8 gUnknown_83CDA94[] = _("/30");
+
```
Comment (jiangzhengwenjz) [inline src/pokemon_storage_system_2.c]: (1 line above) is there any problem on this comment?

Comment (PikalaxALT) [reply src/pokemon_storage_system_2.c]: Yes: making this static will remove the automatic memcpy.

**Pull request merged**